lib/config.js

```diff
@@ -97,8 +98,33 @@ function fromKubeconfig(kubeconfig, current) {
       auth.bearer = user.token;
     }
 
-    if (user['auth-provider'] && user['auth-provider'].config && user['auth-provider'].config['access-token']) {
-      auth.bearer = user['auth-provider'].config['access-token'];
+    if (user['auth-provider']) {
+      const config = user['auth-provider'].config;
+
+      // if we can't determine the type, just fail later (or don't refresh).
+      let type  = null;
+      let token = null;
+      if (config['cmd-path']) {
```

Comment:
Would this and https://github.com/godaddy/kubernetes-client/pull/247/files#diff-f7bf2b665273dfa66ffa4ad7c0a52bb9R111 ever both be true?  if not, can you make that clear using `else if` below?